packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -272,6 +278,20 @@ class AndroidStudio implements Comparable<AndroidStudio> {
       _validationMessages.add(
           'Gradle version $latestGradleVersion at $_gradlePath is not executable.');
     }
+
+    final String javaPath = platform.isMacOS ?
+        fs.path.join(directory, 'jre', 'jdk', 'Contents', 'Home') :
+        fs.path.join(directory, 'jre');
+    final ProcessResult result = processManager.runSync(<String>[fs.path.join(javaPath, 'bin', 'java'), '-version']);
+    if (result.exitCode == 0) {
+      final List<String> versionLines = result.stderr.split('\n');
+      final String javaVersion = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
+      _validationMessages.add('Java version: $javaVersion');
+      _javaPath = javaPath;
+    } else {
+      _isValid = false;
```

Comment:
I'm not entirely sure about this bit, since we have a separate Doctor check for Java. I don't think a missing JRE should invalidate Studio, when you can have a perfectly working system with a separate JDK install.